Posts/2021/04/tfl-IsUserALocalAdministrator.md

```diff
@@ -0,0 +1,130 @@
+```
+
+As you can tell, these cmdlets allow you to add, remove, change, enable and disable a local user or local group
+And they allow you to add, remove and get the local group's members.
+These cmdlets are broadly similar to the ActiveDirectory cmdlets, but work on local users.
+And as noted above, you can use domain users/groups as a member of a local group should you wish or need to.
+
+You use the `Get-LocalGroupMember` command to view the members of a local group, like this:
+
+```powershell-console
+PS> Get-LocalGroupMember -Group 'Administrators'
+
+ObjectClass Name                     PrincipalSource
+----------- ----                     ---------------
+Group       COOKHAM\Domain Admins    ActiveDirectory
+User        COOKHAM24\Administrator  Local
+User        COOKHAM\JerryG           ActiveDirectory
+User        COOKHAM24\Dave           Local
+```
+
+As you can see in this output, the local Administrators group on this host contains domain users and groups as well as local users
+
+## Is the User an Administrator?
+
+It's easy to get membership of any local group, as you saw above.
+But what if you want to find out if a given user is a member of some group?
+Or, determine whether the user running a given script is a member of the administrator's group?
+That too is pretty easy and take a couple of steps.
+One way you can get the name of the current user is by using `whoami.exe`.
+Then you can get the members of the Administrator's group.
+Finally, you check to see if the currently logged on user is a member of the group.
+All of which looks like this:
+
+```powershell-console
+PS> # Get who I am
+PS> $Me = whoami.exe
+PS> $Me 
+Cookham\JerryG
+
+PS> # Get members of administrators group
+PS> $Admins = Get-LocalGroupMember -Name Administrators | 
+       Select-Object -ExpandProperty name
+
+PS> # Check to see if this user is an administrator and act accordingly
+PS> if ($Admins -Contains $me) {
+      "$Me is a local administrator"} 
+    else {
+     "$Me is NOT a local administrator"}
+Cookham\JerryG is a local administrator
+```
+
+If the user running the script, then **$Me** is a user in the local administrator's group.
+And if the current user is in the group, that that means that the logged-on user must be a local administrator.
+
+In this snippet, we just echo the fact that the user is, ir is not, a member of the local administrators group.
+You can adapt it to ensure a user is a member of the appropriate group before attempting to run certain commands.
+And you can also adapt it to check for membership in other local groups such as **Backup Operators** or **Hyper-V Users** which may be relevant.
+
+In your logon script, once you know that the user is a local admin, you can carry out any tasks that require administrative privileges.
+And if the user is not a local or other admin, you could echo that fact, and avoid using a cmdlet that requires the user to be an admin.
```

Comment:
You can only carry out administrator-level tasks if you are running elevated. To find out if you are running elevated:

```powershell
$identity = [Security.Principal.WindowsIdentity]::GetCurrent()
$principal = [Security.Principal.WindowsPrincipal] $identity
$principal.IsInRole([Security.Principal.WindowsBuiltInRole] 'Administrator')
```
`$principal.IsInRole()` returns a boolean to tell you if you are running elevated.

Reply:
Agreed. But we are talking logon scripts. So I have changed the thrust a bit - looking at running commands that require membership of some group.

Reply:
a great point. Since the logon script doesn't, as such, run as elevated this is not relevant in this case. So I have refocused the post to being a member of a group.